code/obj/item/device/studio_monitor.dm

```diff
@@ -199,6 +199,15 @@
 			boutput(src.the_mob, "<span class='alert'>You are already playing something...</span>")
 			. = FALSE
 
+	proc/is_rock_immune(mob/living/target)
+		if(isvirtual(target))
+			var/mob/living/carbon/human/H
+			if(istype(target, /mob/living/carbon/human))
```

Comment:
why is this double istype here and too high of a cast (not to human/virtual)

Reply:
_excuses regarding being functional_
_agreeing upon more maintainable implementation_